docs/visual-basic/language-reference/modifiers/shared.md

```diff
@@ -13,84 +13,85 @@ helpviewer_keywords:
-  
-     The following example calls the shared procedure <xref:System.Double.IsNaN%2A> exposed by the <xref:System.Double> structure.  
-  
-     `If Double.IsNaN(result) Then MsgBox("Result is mathematically undefined.")`  
-  
-- **Implicit Sharing.** You cannot use the `Shared` modifier in a [Const Statement](../../../visual-basic/language-reference/statements/const-statement.md), but constants are implicitly shared. Similarly, you cannot declare a member of a module or an interface to be `Shared`, but they are implicitly shared.  
-  
-## Behavior  
-  
-- **Storage.** A shared variable or event is stored in memory only once, no matter how many or few instances you create of its class or structure. Similarly, a shared procedure or property holds only one set of local variables.  
-  
-- **Accessing through an Instance Variable.** It is possible to access a shared element by qualifying it with the name of a variable that contains a specific instance of its class or structure. Although this usually works as expected, the compiler generates a warning message and makes the access through the class or structure name instead of the variable.  
-  
-- **Accessing through an Instance Expression.** If you access a shared element through an expression that returns an instance of its class or structure, the compiler makes the access through the class or structure name instead of evaluating the expression. This produces unexpected results if you intended the expression to perform other actions as well as returning the instance. The following example illustrates this.  
+
+Specifies that one or more declared programming elements are associated with a class or structure at large, and not with a specific instance of the class or structure.
+
+## When to Use Shared
+
+Sharing a member of a class or structure makes it available to every instance, rather than *non-shared*, where each instance keeps its own copy. This is useful, for example, if the value of a variable applies to the entire application. If you declare that variable to be `Shared`, then all instances access the same storage location, and if one instance changes the variable's value, all instances access the updated value.
+
+Sharing does not alter the access level of a member. For example, a class member can be shared and private (accessible only from within the class), or non-shared and public. For more information, see [Access levels in Visual Basic](../../../visual-basic/programming-guide/language-features/declared-elements/access-levels.md).
+
+## Rules
+
+- **Declaration Context.** You can use `Shared` only at module level. This means the declaration context for a `Shared` element must be a class or structure, and cannot be a source file or namespace.
```

Comment:
I removed "procedure" here. It looks wrong (or misleading if it meant something else) because a Sub or Function can be Shared.

Reply:
The word "procedure" should stay. It's qualified by "declaration context", so it means a variable declared in a procedure cannot be shared.

```suggestion
- **Declaration Context.** You can use `Shared` only at module level. This means the declaration context for a `Shared` element must be a class or structure, and cannot be a source file, namespace, or procedure.
```